src/App.jsx

```diff
@@ -1,22 +1,32 @@
-import React, { useState } from "react";
-import TodoList from "./TodoList";
-import AddTodoForm from "./AddTodoForm";
- 
+   
+   // eslint-disable-next-line no-unused-vars
+   import React, { useEffect, useState } from "react";
+    import TodoList from "./TodoList";
+    import AddTodoForm from "./AddTodoForm";
+        function useSemiPersistentState (){
+        const [todoList, setTodoList] = useState(() => {
+        const savedTodoList = localStorage.getItem ('savedTodoList');
+            return  savedTodoList ? JSON.parse (savedTodoList) : [];
```

Comment:
I like how you're thinking about the case when localStorage does not yet have a value for savedTodoList and returning an empty list as a default value. Nice! However this could be simplified and if localStorage.getItem('savedTodoList') returns null, JSON.parse(null) will throw an error. You can handle this by providing a default value directly in JSON.parse:
```
const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList') || '[]'));
```